test/cypress/integration/stripe-setup/vat-number-company-number/check_your_answers_spec.js

```diff
@@ -0,0 +1,334 @@
+          stubStripeAccountGet(gatewayAccountId, 'acct_123example123')
+        ])
+
+        cy.setEncryptedCookies(userExternalId, gatewayAccountId, {})
+      })
+
+      it('should display page correctly with company number populated', () => {
+        cy.visit('/vat-number-company-number/vat-number')
+        cy.get('#vat-number-form').should('exist')
+          .within(() => {
+            cy.get('input#vat-number[name="vat-number"]').type(validVatNumber)
+
+            cy.get('button[type=submit]').click()
+          })
+
+        cy.location().should((location) => {
+          expect(location.pathname).to.eq('/vat-number-company-number/company-number')
+        })
+        cy.get('#company-number-form').should('exist')
+          .within(() => {
+            cy.get('input#company-number-mode-1[name="company-number-mode"]').check()
+            cy.get('input#company-number[name="company-number"]').type(validCompanyNumber)
+
+            cy.get('button[type=submit]').click()
+          })
+
+        cy.location().should((location) => {
+          expect(location.pathname).to.eq('/vat-number-company-number/check-your-answers')
+        })
+        cy.get('dl.govuk-summary-list > div.govuk-summary-list__row:nth-child(1) > dd.govuk-summary-list__value')
+          .should('contain', validVatNumber)
+        cy.get('dl.govuk-summary-list > div.govuk-summary-list__row:nth-child(2) > dd.govuk-summary-list__value')
+          .should('contain', validCompanyNumber)
+      })
+
+      it('should display page correctly with company number not populated', () => {
+        cy.visit('/vat-number-company-number/vat-number')
+        cy.get('#vat-number-form').should('exist')
+          .within(() => {
+            cy.get('input#vat-number[name="vat-number"]').type(validVatNumber)
+
+            cy.get('button[type=submit]').click()
+          })
+
+        cy.location().should((location) => {
+          expect(location.pathname).to.eq('/vat-number-company-number/company-number')
+        })
+        cy.get('#company-number-form').should('exist')
+          .within(() => {
+            cy.get('input#company-number-mode-2[name="company-number-mode"]').check()
+            cy.get('input#company-number[name="company-number"]').should('not.be.visible')
+
+            cy.get('button[type=submit]').click()
+          })
+
+        cy.location().should((location) => {
+          expect(location.pathname).to.eq('/vat-number-company-number/check-your-answers')
+        })
+        cy.get('dl.govuk-summary-list > div.govuk-summary-list__row:nth-child(1) > dd.govuk-summary-list__value')
+          .should('contain', validVatNumber)
+        cy.get('dl.govuk-summary-list > div.govuk-summary-list__row:nth-child(2) > dd.govuk-summary-list__value')
+          .should('contain', 'None')
+      })
+
+      it('should go to VAT number page when VAT number "change" button is clicked and change tbe value', () => {
```

Comment:
To avoid having to go through all the steps each time for these happy path tests with a common setup, which takes time, is there an argument that they could be in a single describe along with the first happy path test using `Cypress.Cookies.preserveOnce(names...)` to keep the cookies between the different "if"s.
This way you could combine the checking the display of the check your answers page and checking going back and changing the answers in one continuing describe block.

This would also save a lot of lines of code. What do you think?

Reply:
Although given time constraints I wouldn't bother changing it. Maybe an improvement for someone else to do in the future...

Reply:
Yes, great idea! I was also not happy to copy & pasta a lot of code, `Cypress.Cookies.preserveOnce` is a good approach as you suggested.
As discussed this can be done later at some point.